Libplanet/Net/Swarm.cs

```diff
@@ -33,37 +35,43 @@ public partial class Swarm : ICollection<Peer>, IDisposable
         private readonly RouterSocket _router;
         private readonly IDictionary<Address, DealerSocket> _dealers;
 
-        private readonly Uri _listenUrl;
         private readonly TimeSpan _dialTimeout;
         private readonly AsyncLock _distributeMutex;
         private readonly AsyncLock _receiveMutex;
+        private readonly AsyncLock _blockSyncMutex;
+        private readonly IPAddress _ipAddress;
 
         private readonly ILogger _logger;
 
         private TaskCompletionSource<object> _runningEvent;
+        private int? _listenPort;
 
         public Swarm(
             PrivateKey privateKey,
-            Uri listenUrl,
-            int millisecondsDialTimeout,
+            int millisecondsDialTimeout = 15000,
+            IPAddress ipAddress = null,
+            int? listenPort = null,
             DateTimeOffset? createdAt = null)
             : this(
                   privateKey,
-                  listenUrl,
                   TimeSpan.FromMilliseconds(millisecondsDialTimeout),
+                  ipAddress,
+                  listenPort,
                   createdAt)
         {
         }
 
         public Swarm(
             PrivateKey privateKey,
-            Uri listenUrl,
-            TimeSpan? dialTimeout = null,
+            TimeSpan dialTimeout,
+            IPAddress ipAddress = null,
+            int? listenPort = null,
             DateTimeOffset? createdAt = null)
         {
-            _privateKey = privateKey;
-            _listenUrl = listenUrl;
-            _dialTimeout = dialTimeout ?? TimeSpan.FromMilliseconds(15000);
+            _runningEvent = new TaskCompletionSource<object>();
```

Comment:
This seems redundant; this is initialized at the bottom of the constructor:

```c#
Running = false;
```

This property invokes its setter, which creating a new instance of `TaskCompletionSource<object>`:

```c#
public bool Running
{
    get => _runningEvent.Task.Status == TaskStatus.RanToCompletion;

    private set
    {
        if (value)
        {
            _runningEvent.TrySetResult(null);
        }
        else
        {
            _runningEvent = new TaskCompletionSource<object>();
        }
    }
}
```

Reply:
The exact reason is unknown, but it seems that the destructor is called before calling the setter that you talked about. (#97) so I've put this line here as workaround.

But, moving set timing and eliminating redundant is an orthogonal problem, so I'll put `Running = false` here.